### PR TITLE
Test with MRI 2.3.1

### PR DESCRIPTION
**Pull request by bquorning**
New Ruby loads many new `IO` methods.

### CODE REVIEW

File: test/fakefs_test.rb [hunk 1]
```diff
@@ -1262,7 +1262,16 @@ def test_should_set_pos_with_tell_method
...
+    :pathconf,
+    :cursor, :cursor=,
+    :beep, :goto, :pressed?,
+
+    # omit methods from io/nonblock
+    :nonblock, :nonblock=, :nonblock?,
+
+    # omit methods from io/wait
+    :nread, :ready?,
+    :wait, :wait_readable, :wait_writable
```
Comment (grosser) [inline test/fakefs_test.rb]: does it fail without these ... or why are they being added ?

**Comment by grosser** [comment]
can you rebase ?

**Comment by bquorning** [comment]
I added the symbols needed to make it pass with 2.3.1.

Will rebase later.

File: test/fakefs_test.rb (hunk 1 above)
Comment (bquorning) [inline test/fakefs_test.rb]: Hmm, it seems that `RealFile` isn’t just the original `File` at this point in the tests – it has gained a lot of extra methods…

I’ll investigate a bit more.

Comment (bquorning) [inline test/fakefs_test.rb]: RuboCop requires `ruby-progressbar` which requires `io/console`:
```
» ruby -e "x = File.instance_methods ; require 'ruby-progressbar' ; p File.instance_methods - x"
[:winsize, :raw, :raw!, :cooked, :cooked!, :getch, :echo=, :echo?, :noecho, :winsize=, :iflush, :oflush, :goto, :cursor, :ioflush, :beep, :getpass, :cursor=, :pressed?]
```
RuboCop also requires `net/http`:
```
» ruby -e "x = File.instance_methods ; require 'net/http' ; p File.instance_methods - x"
[:wait, :wait_readable, :wait_writable, :nread, :ready?]
```

Comment (grosser) [inline test/fakefs_test.rb]: bah ... this is annoying ...

Comment (bquorning) [inline test/fakefs_test.rb]: Well, even if I remove RuboCop from the gemspec, those methods are there.

Comment (grosser) [inline test/fakefs_test.rb]: just do whatever is needed :D 
... or is there any other dynamic way to solve that ?

**Comment by bquorning** [comment]
Interesting:
```
» cat Gemfile.lock
GEM
  specs:

PLATFORMS
  ruby

DEPENDENCIES

BUNDLED WITH
   1.13.2
» ruby -e "puts RUBY_VERSION"
2.3.1
» ruby -e "puts File.instance_methods.size"
190
» bundle exec ruby -e "puts File.instance_methods.size"
219
» chruby 2.2
» ruby -e "puts RUBY_VERSION"
2.2.4
» ruby -e "puts File.instance_methods.size"
188
» bundle exec ruby -e "puts File.instance_methods.size"
204
```
So in Ruby 2.2, bundler adds 16 methods to `File`, while in Ruby 2.3, bundler adds 29 methods to `File`.

**Comment by bquorning** [comment]
bundler loads rubygems, and the rubygems implementation differs by Ruby version. And the libraries required by rubygems also differ by Ruby version:

2.2:
```rb
» ruby -e "x = File.instance_methods ; require 'resolv' ; p File.instance_methods - x"
[]
```

2.3
```rb
» ruby -e "x = File.instance_methods ; require 'resolv' ; p File.instance_methods - x"
[:wait, :nread, :ready?, :nonblock=, :wait_readable, :wait_writable, :nonblock, :nonblock?]
```

I think I’ll just add all the `io/whatever` methods to the `OMITTED_FILE_METHODS` array.

**Comment by bquorning** [comment]
Green build; except for JRuby and Rubinius.

**Comment by grosser** [comment]
🎉

**Pull request merged**